src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -406,7 +435,33 @@ private function createFirewall(ContainerBuilder $container, string $id, array $
         $configuredEntryPoint = isset($firewall['entry_point']) ? $firewall['entry_point'] : null;
 
         // Authentication listeners
-        list($authListeners, $defaultEntryPoint) = $this->createAuthenticationListeners($container, $id, $firewall, $authenticationProviders, $defaultProvider, $providerIds, $configuredEntryPoint, $contextListenerId);
+        $firewallAuthenticationProviders = [];
+        list($authListeners, $defaultEntryPoint) = $this->createAuthenticationListeners($container, $id, $firewall, $firewallAuthenticationProviders, $defaultProvider, $providerIds, $configuredEntryPoint, $contextListenerId);
+
+        $authenticationProviders = array_merge($authenticationProviders, $firewallAuthenticationProviders);
+
+        if ($this->authenticatorManagerEnabled) {
+            // authenticator manager
+            $authenticators = array_map(function ($id) {
+                return new Reference($id);
+            }, $firewallAuthenticationProviders);
+            $container
+                ->setDefinition($managerId = 'security.authenticator.manager.'.$id, new ChildDefinition('security.authenticator.abstract_manager'))
+                ->replaceArgument(0, $authenticators)
+                ->replaceArgument(2, new Reference($firewallEventDispatcherId))
+                ->replaceArgument(3, $id)
+                ->addTag('security.authenticator_manager', ['firewall' => $id])
```

Comment:
This is a nit-pick. This is to make it easy to pass a `tagged_locator` to `FirewallAwareAuthenticatorManager`. But this is not a tag that a *user* should ever try to use to "push" another "authenticator manager" into that class. And so, it would probably be better if we set that argument directly (without the tag)